### PR TITLE
Changes for the proposal

### PR DESCRIPTION
**Pull request by mitzkia**
- I have created the basic changes to fit with the proposal in https://github.com/balabit/syslog-ng/issues/573 . 
- This change provides the following syslog-ng docker images (it is only support the latest syslog-ng version):
  - balabit/syslog-ng:latest
  - balabit/syslog-ng-dev:latest
  - balabit/syslog-ng-incubator-dev:latest
- After this change you can use the following commands to use balabit/syslog-ng and balabit/syslog-ng-dev images:
  - sudo docker run balabit/syslog-ng:latest
  - sudo docker run -it -v $HOME/syslog-ng:/source balabit/syslog-ng-dev:latest /bin/bash
  - sudo docker run -it -v $HOME/syslog-ng-incubator:/incubator-source balabit/syslog-ng-incubator-dev:latest /bin/bash
- With this change we support only Debian Jessie OS.
- The change also contains a Readme.md which describes the available docker images and how to use them in certain cases.

### CODE REVIEW

**Comment by ihrwein** [comment]
Laci and me reviewed only the Dockerfiles. You can find our comments attached to the second commit.

I'll review the documentation Monday. Thanks you very much!

**Comment by mitzkia** [comment]
Thank you for the review notes. 
I fixed your findings.

File: Dockerfiles/syslog-ng-dev/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+    wget
+
+RUN wget -qO - http://download.opensuse.org/repositories/home:/laszlo_budai:/syslog-ng-3.6/Debian_8.0/Release.key | apt-key add -
+RUN echo 'deb http://download.opensuse.org/repositories/home:/laszlo_budai:/syslog-ng-3.6/Debian_8.0 ./' | tee --append /etc/apt/sources.list.d/syslog-ng-obs.list
+
+ADD dev-dependencies.txt .
+RUN cat dev-dependencies.txt | grep -v "#" | xargs apt-get install -y
+
+ADD openjdk-libjvm.conf /etc/ld.so.conf.d/openjdk-libjvm.conf
+ADD libsyslog-ng.conf /etc/ld.so.conf.d/libsyslog-ng.conf
```
Comment (ihrwein) [inline Dockerfiles/syslog-ng-dev/Dockerfile]: don't we need to execute `ldconfig` after these lines?

**Comment by ihrwein** [comment]
If you update the PR please don't force push the branch but use small fixing commits. When there is nothing to fix I'll ask you to squash them into 2 commits and after that it will be merged :)

File: Dockerfiles/syslog-ng-dev/dev-dependencies.txt [hunk 2]
```diff
@@ -16,6 +21,7 @@ libhiredis-dev
 libnet1-dev
 libpython-dev
 libssl-dev
+libriemann-client-dev
 libsystemd-daemon-dev
```
Comment (ihrwein) [inline Dockerfiles/syslog-ng-dev/dev-dependencies.txt]: I've checked this package and it's only a transitional package. It directly depends on `libsystemd-dev`. So could you change it?

File: Dockerfiles/syslog-ng-incubator-dev/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    cd /root/grok && \
+    autoreconf -i && \
+    ./configure && \
+    make && \
+    make install
+
+ADD pip-dependencies.txt .
+RUN pip install -r pip-dependencies.txt
+
+RUN mkdir /incubator-source
```
Comment (ihrwein) [inline Dockerfiles/syslog-ng-incubator-dev/Dockerfile]: Just a note here, I sent you a mail about this.

File: Dockerfiles/syslog-ng-incubator-dev/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+    syslog-ng-dev
+
+RUN git clone https://github.com/talien/grok.git /root/grok && \
+    cd /root/grok && \
+    autoreconf -i && \
+    ./configure && \
+    make && \
+    make install
+
+ADD pip-dependencies.txt .
```
Comment (ihrwein) [inline Dockerfiles/syslog-ng-incubator-dev/Dockerfile]: ~~I'd call this simply `requirements.txt`.  We don't have a lot of files and this name is the convention.~~

Does incubator really need python packages installed?

File: Dockerfiles/syslog-ng/syslog-ng.conf [hunk 5]
```diff
@@ -0,0 +1,18 @@
+@version: 3.6
+
+source s_net {
+  tcp(
+     ip("127.0.0.1")
+     port(1111)
```
Comment (ihrwein) [inline Dockerfiles/syslog-ng/syslog-ng.conf]: Why the port 1111? I'd use a syslog source or a tcp without ports (default ports should do).

File: Dockerfiles/syslog-ng/syslog-ng.conf [hunk 6]
```diff
@@ -0,0 +1,18 @@
+@version: 3.6
+
+source s_net {
+  tcp(
+     ip("127.0.0.1")
```
Comment (ihrwein) [inline Dockerfiles/syslog-ng/syslog-ng.conf]: I think we should use the `0.0.0.0` address to listen on. AFAIK we can't reach the service from outside if it only listens on the loopback address.

File: Readme.md [hunk 7]
```diff
@@ -0,0 +1,136 @@
+## What kind of syslog-ng docker images are exist
+
+  * There are two types of docker images:
+     * balabit/syslog-ng image where syslog-ng is installed in a fresh Debian 8 (Jessie) operation system from (official) deb package. This docker image is for to use syslog-ng as system logging service.
+     * balabit/syslog-ng-dev development image. This image based on also a fresh Debian 8 (Jessie). This image is not contain syslog-ng, it only contains all packages to build and debug syslog-ng within the container.
```
Comment (ihrwein) [inline Readme.md]: I'd remove `from (official) deb package`. I don't think the users are interested in it and the `official` word maybe a little strong.

**Comment by ihrwein** [comment]
I've sent you a PR regarding the documentation. After that I'd move the `syslog-ng`, `syslog-ng-dev` and `syslog-ng-incubator-dev` directories into the root of the repo (so we wouldn't need the `Dockerfiles` directory).

**Comment by mitzkia** [comment]
I merged your PR. 
Squashed Readme changes into one commit. 
Creates a new commit about moving subdirectories into the root directory.

**Comment by ihrwein** [comment]
Thanks!

@lbudai Could you review the PR, too?

**Comment by lbudai** [comment]
:+1:

**Comment by ihrwein** [comment]
Thanks!

**Pull request merged**